src/Microsoft.AspNet.Routing/Template/TemplateMatcher.cs

```diff
@@ -67,7 +63,7 @@ public class TemplateMatcher
                     var part = routeSegment.Parts[0];
                     if (part.IsLiteral)
                     {
-                        if (!string.Equals(part.Text, requestSegment, StringComparison.OrdinalIgnoreCase))
+                        if (!requestSegment.Equals(part.Text))
```

Comment:
This now does a comparison against a range on the original string and is allocation free. Since this is a literal segment, we never need to substring this.